api/api.go

```diff
@@ -180,3 +190,26 @@ func handleHealthcheck(c *gin.Context) {
 	// TODO: add some actual health checking here
 	c.JSON(200, gin.H{"failed": false})
 }
+
+func findProjectRoot() (string, error) {
+	// Start with the directory of the current file or test.
+	dir, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	// Walk up the directory tree looking for the marker.
+	for {
+		if _, err := os.Stat(filepath.Join(dir, ".projectroot")); err == nil {
+			return dir, nil
+		}
+
+		// Move to the parent directory.
+		nextDir := filepath.Dir(dir)
+		if nextDir == dir {
+			// We've reached the root of the filesystem.
+			return "", fmt.Errorf("project root not found")
+		}
+		dir = nextDir
+	}
+}
```

Comment:
We probably don't want to do anything within snek that expects the original source repo.